server/mrt.go

```diff
@@ -88,11 +88,17 @@ func (m *mrtWriter) loop() error {
 			msg := make([]*mrt.MRTMessage, 0, 1)
 			switch e := ev.(type) {
 			case *WatchEventUpdate:
-				subtype := mrt.MESSAGE_AS4
 				mp := mrt.NewBGP4MPMessage(e.PeerAS, e.LocalAS, 0, e.PeerAddress.String(), e.LocalAddress.String(), e.FourBytesAs, nil)
 				mp.BGPMessagePayload = e.Payload
-				if e.FourBytesAs == false {
-					subtype = mrt.MESSAGE
+				isAddPaths := e.PathList[0].GetNlri().PathIdentifier() != 0
```

Comment:
Thanks, I misunderstood. If zero is valid, there is no reserved path_id (can check path_id is advertised or not), I guess.
Instead of path_id, I will fix to check if the neighbor has add-paths feature enabled for the given family or not.